### DISCUSSION TITLE
0088. 合并两个有序数组

### DISCUSSION THREAD

**Discussion by giscus** [itcharge/AlgoNote#151] [Comments]
# 0088. 合并两个有序数组

[https://algo.itcharge.cn/solutions/0001-0099/merge-sorted-array/](https://algo.itcharge.cn/solutions/0001-0099/merge-sorted-array/)

**Comment by ccczy-czy** [comment]
为什么指针都要指向尾部呢？

**Reply by Therm4l** [reply]
我也想过这个问题，我认为是因为如果从前往后进行覆盖的话，会把需要用到的值给覆盖掉。就比如`nums1[0]`和`nums2[0]`进行比较，如果`nums2[0]`的值更小并覆上去的话，原先`nums1[0]`的值就无法保留。

但是如果从后往前就不会有这个问题，因为是首先比较`nums1[m-1]`和`nums2[n-1]`，然后将较大的元素放入`nums1[m+n-1]`。这样，nums1中未处理的元素（即`nums1[0]`到`nums1[m-1]`）就不会被覆盖。

**Comment by Panda7n7** [comment]
前面可以加一个 index1指向nums1的第m个元素尾部，出现歧义指向数组尾部
